d1985r3.md

```diff
@@ -222,6 +222,78 @@ OK, but and treating them like this also eases specification, and is more
 consistent with the way the language currently works. This is the more
 "adventurous" option, however.
 
+### Dependent names
+
+Universal template parameters are always dependent names. When a
+universal template parameter is used in an expression, it is parsed as a
+value by default. They can be disambiguated as a type by using `typename`
+as with other dependent names. To defer the type/value decision, `template
+auto` maybe used to perserve the name as a universal template. This allows
+the universal template parameter to be forwarded to another template.
+
+```cpp
+template<template auto Arg> void f()
+{
+  std::cout << Arg; // Arg is parsed a value
+  std::cout << sizeof(typename Arg); // Arg is parsed as a type
+
+  // Type/value decision is deferred to myclass
+  std::cout << myclass<template auto Arg>::value;
+}
+```
+
+The `typename auto` specifier may also be used to defer type/value lookup and
+form universal template parameters from other dependent names. This enables
+utility structs to perform transformations on packs of universal template
+parameters:
+
+
+```cpp
+template<typename T>
+struct unwrap_nttps;
+
+template<typename T>
+struct unwrap_nttps<T>
+{
+  using result = T;
+};
+
+template<typename T, T t>
+struct unwrap_nttps<std::integral_constant<T,t>>
+{
+  static inline constexpr T result = t;
+};
+```
+
+`template auto unwrap_nttps<Pack>::result...` may result in a template parameter
```

Comment:
template argument pack, right?

Reply:
I believe "template parameter pack" is the correct name for this. It's used as an argument, but "parameter pack" is what the pack itself is called.